### PR TITLE
[Cookbook] Make registration_form follow best practices

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | - |

This is just a rebase of #4773 to make it mergeable.

### CODE REVIEW

File: cookbook/doctrine/registration_form.rst [hunk 1]
```diff
@@ -74,16 +75,12 @@ With some validation added, your class may look something like this::
...
-        private $plainPassword;
-
-        /**
          * The below length depends on the "algorithm" you use for encoding
          * the password, but this works well with bcrypt
          *
-         * @ORM\Column(type="string", length=64)
+         * @ORM\Column(type="string", length=255)
+         * @Assert\NotBlank()
+         * @Assert\Length(max=4096)
```
Comment (xabbuh) [inline cookbook/doctrine/registration_form.rst]: This is useless imo as we do not store the plain password.

File: cookbook/doctrine/registration_form.rst [hunk 2]
```diff
@@ -135,7 +132,7 @@ example, see the :ref:`Entity Provider <security-crete-user-entity>` article.
 
 .. sidebar:: Why the 4096 Password Limit?
 
-    Notice that the ``plainPassword`` field has a max length of 4096 characters.
+    Notice that the ``password`` field has a max length of 4096 characters.
```
Comment (xabbuh) [inline cookbook/doctrine/registration_form.rst]: This whole paragraph must be reworded to reflect the fact that we only store the hashed (encrypted) password.

File: cookbook/doctrine/registration_form.rst [hunk 3]
```diff
@@ -224,8 +223,7 @@ controller for displaying the registration form::
             $form->handleRequest($request);
             if ($form->isSubmitted() && $form->isValid()) {
                 // 3) Encode the password (you could also do this via Doctrine listener)
-                $encoder = $this->get('security.encoder_factory')
-                    ->getEncoder($user);
+                $encoder = $this->get('security.encoder_factory')->getEncoder($user);
                 $password = $encoder->encodePassword($user->getPlainPassword(), $user->getSalt());
```
Comment (xabbuh) [inline cookbook/doctrine/registration_form.rst]: The form handling doesn't work as is as we do not have a property for the plain password in the registration form. I think we should use a DTO class to back the form instead.

**Comment by javiereguiluz** [comment]
I agree with the changes asked by @xabbuh, but they would mean a lot of modifications in the current article. That's why I'm closing this pull request.

**Comment by xabbuh** [comment]
Alright, I opened #4773 again to not forget about this.

**Pull request closed without merging**